netty/src/main/java/io/grpc/netty/ProtocolNegotiators.java

```diff
@@ -103,6 +109,47 @@ public AsciiString scheme() {
   }
 
   /**
+   * Create a server plaintextUpgrade handler for gRPC.
+   */
+  public static ProtocolNegotiator serverPlaintextUpgrade() {
+    return new ProtocolNegotiator() {
+      public Handler newHandler(final GrpcHttp2ConnectionHandler handler) {
+        class PlaintextUpgradeHandler extends ChannelHandlerAdapter implements Handler {
+          PlaintextUpgradeHandler() {
+          }
+
+          public void handlerAdded(ChannelHandlerContext ctx) throws Exception {
+
+            final HttpServerUpgradeHandler.UpgradeCodecFactory upgradeCodecFactory = new HttpServerUpgradeHandler.UpgradeCodecFactory() {
+              @Override
+              public UpgradeCodec newUpgradeCodec(CharSequence protocol) {
+                if (AsciiString.contentEquals(Http2CodecUtil.HTTP_UPGRADE_PROTOCOL_NAME, protocol)) {
+                    return new Http2ServerUpgradeCodec(handler);
+                  } else {
+                    return null;
+                  }
+              }
+            };
+
+            HttpServerCodec sourceCodec = new HttpServerCodec();
+            HttpServerUpgradeHandler upgradeCodec = new HttpServerUpgradeHandler(sourceCodec, upgradeCodecFactory);
+            CleartextHttp2ServerUpgradeHandler upgrader = new
```

Comment:
Word-wrap before `new`, not after (so keep `new CleartextHttp2ServerUpgradeHandler(` together on the same line).